app/src/main/java/org/p2p/wallet/root/RootPresenter.kt

```diff
@@ -21,6 +27,15 @@ class RootPresenter(
         }
     }
 
+    override fun loadAvailableNetworkEnvironments() {
+        environmentManager.loadAvailableEnvironments(networkListFeatureToggle.value.toDomain())
+    }
+
+    private fun List<SettingsNetworkValue>.toDomain(): List<NetworkEnvironment> {
+        val allNetworks = NetworkEnvironment.values()
+        return mapNotNull { toggleNetwork -> allNetworks.find { it.endpoint == toggleNetwork.url } }
```

Comment:
weird mapping. 
Why you just check url instead of passing all params?

Reply:
oh, correct, i will replace it with `filter`